src/analysis/inner_product_space/linear_pmap.lean

```diff
@@ -0,0 +1,216 @@
+
+We use the junk value pattern to define the adjoint for all `linear_pmap`s. In the case that
+`T : E →ₗ.[𝕜] F` is not densely defined the adjoint `T†` is the zero map from `T.adjoint_domain` to
+`E`.
+
+## References
+
+* [J. Weidmann, *Linear Operators in Hilbert Spaces*][weidmann_linear]
+
+## Tags
+
+Unbounded operators, closed operators
+-/
+
+
+noncomputable theory
+
+open is_R_or_C
+open_locale complex_conjugate classical
+
+variables {𝕜 E F G : Type*} [is_R_or_C 𝕜]
+variables [normed_add_comm_group E] [inner_product_space 𝕜 E]
+variables [normed_add_comm_group F] [inner_product_space 𝕜 F]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 _ _ x y
+
+namespace linear_pmap
+
+/-- An operator `T` is a formal adjoint of `S` if for all `x` in the domain of `T` and `y` in the
+domain of `S`, we have that `⟪T x, y⟫ = ⟪x, S y⟫`. -/
+def is_formal_adjoint (T : E →ₗ.[𝕜] F) (S : F →ₗ.[𝕜] E) : Prop :=
+∀ (x : T.domain) (y : S.domain), ⟪T x, y⟫ = ⟪(x : E), S y⟫
+
+variables {T : E →ₗ.[𝕜] F} {S : F →ₗ.[𝕜] E}
+
+@[protected] lemma is_formal_adjoint.symm (h : T.is_formal_adjoint S) : S.is_formal_adjoint T :=
+λ y _, by rw [←inner_conj_symm, ←inner_conj_symm (y : F), h]
+
+variables (T)
+
+/-- The domain of the adjoint operator.
+
+This definition is needed to construct the adjoint operator and the preferred version to use is
+`T.adjoint.domain` instead of `T.adjoint_domain`. -/
+def adjoint_domain : submodule 𝕜 F :=
+{ carrier := {y | continuous ((innerₛₗ 𝕜 y).comp T.to_fun)},
+  zero_mem' := by { rw [set.mem_set_of_eq, linear_map.map_zero, linear_map.zero_comp],
+      exact continuous_zero },
+  add_mem' := λ x y hx hy, by { rw [set.mem_set_of_eq, linear_map.map_add] at *, exact hx.add hy },
+  smul_mem' := λ a x hx, by { rw [set.mem_set_of_eq, linear_map.map_smulₛₗ] at *,
+    exact hx.const_smul (conj a) } }
+
+/-- The operator `λ x, ⟪y, T x⟫` considered as a continuous linear operator from `T.adjoint_domain`
+to `𝕜`. -/
+def adjoint_domain_mk_clm (y : T.adjoint_domain) : T.domain →L[𝕜] 𝕜 :=
+⟨(innerₛₗ 𝕜 (y : F)).comp T.to_fun, y.prop⟩
+
+lemma adjoint_domain_mk_clm_apply (y : T.adjoint_domain) (x : T.domain) :
+  adjoint_domain_mk_clm T y x = ⟪(y : F), T x⟫ := rfl
+
+variable {T}
+variable (hT : dense (T.domain : set E))
+
+include hT
+
+/-- The unique continuous extension of the operator `adjoint_domain_mk_clm` to `E`. -/
+def adjoint_domain_mk_clm_extend (y : T.adjoint_domain) :
+  E →L[𝕜] 𝕜 :=
+(T.adjoint_domain_mk_clm y).extend (submodule.subtypeL T.domain)
+  hT.dense_range_coe uniform_embedding_subtype_coe.to_uniform_inducing
+
+@[simp] lemma adjoint_domain_mk_clm_extend_apply (y : T.adjoint_domain) (x : T.domain) :
+  adjoint_domain_mk_clm_extend hT y (x : E) = ⟪(y : F), T x⟫ :=
+continuous_linear_map.extend_eq _ _ _ _ _
+
+variables [complete_space E]
+
+/-- The image of the adjoint operator.
+
+This is an auxiliary definition needed to define the adjoint operator as a `linear_pmap`. -/
+def adjoint_elem (y : T.adjoint_domain) : E :=
```

Comment:
Is your intent that this definition should not be used outside this file because it breaks the API barrier? If so, you should make that clear, and you should also put the appropriate warning on any of the lemmas that break that API barrier. If not, I would rephrase to avoid the word *auxiliary*.

Reply:
Deleted that definition, it got useless after Eric's suggestion.